dart/common/NameManager.h

```diff
@@ -189,6 +227,12 @@ class NameManager
     return (mMap.find(_name) != mMap.end());
   }
 
+  /// Return true if the object is contained
+  bool hasObject(T _obj) const
+  {
+    return (mReverseMap.find(_obj) != mReverseMap.end());
+  }
+
   /// Get the number of the objects currently stored by the NameManager
   size_t getCount() const
   {
```

Comment:
getCount() can't actually be overloaded since it's not virtual, so nothing to worry about there.

Just checking the sizes shouldn't be expensive at all. I was imagining running through the maps and checking for a one-to-one correspondence on their entries. I'd have no objections to just a quick size comparison.